docling_core/types/newdoc/document.py

```diff
@@ -0,0 +1,95 @@
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from pydantic import AnyUrl, BaseModel, Field
+
+from docling_core.types.newdoc.base import BoundingBox, Size
+
+
+class FigureData(BaseModel):  # TBD
+    pass
+
+
+class TableData(BaseModel):  # TBD
+    pass
+
+
+class RefItem(BaseModel):
+    cref: str = Field(alias="$ref")
+
+    def resolve(self, doc: "DoclingDocument"):
+        _, path, index_str = self.cref.split("/")
+        index = int(index_str)
+        obj = doc.__getattribute__(path)[index]
+        return obj
+
+
+class ImageRef(BaseModel):
+    format: str  # png, etc.
+    dpi: int  # ...
+    size: Size
+    uri: AnyUrl
+
+
+class ProvenanceItem(BaseModel):
+    page_no: int
+    bbox: BoundingBox
+    charspan: Tuple[int, int]
+
+
+class DocItem(BaseModel):
+    dloc: str  # format spec ({document_hash}{json-path})
+    hash: int
```

Comment:
Can we specify explicitly,

from pydantic import BaseModel, conint

# Define a constrained integer for uint64
Uint64 = conint(ge=0, le=(2**64 - 1))

class DocItem(BaseModel):
    hash: Uint64